src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,75 @@
+
+package codeu.chat.util;
+
+import java.io.*;
+
+public final class Tokenizer {
+
+    public Tokenizer (String source) {
+        this.source = source;
+        at = 0;
+    }
+
+    public String next() throws IOException {
+        // Skip all leading whitespace
+        while (remaining() > 0 && Character.isWhitespace(peek())) {
+            read();  // ignore the result because we already know that it is a whitespace character
```

Comment:
In-line comments are nice and clear. They are not too frequent and they make the code very readable.